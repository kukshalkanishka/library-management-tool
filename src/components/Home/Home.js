import React from "react";
import AddBookPrompt from "../AddTodoPrompt/AddBookPrompt";
import BookList from "../BookList/BookList";

class Home extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      books: {}
    };
  }

  async loadBooks() {
    await fetch("/todos", { method: "POST" })
      .then(res => res.json())
      .then(res => {
        console.log(res, ">>>>>>>>>>>>>>>");
        this.setState({ books: res.groupedBooks });
      });
  }

  async componentDidMount() {
    await this.loadBooks();
  }

  render() {
    return (
      <div className="todoMainContainer">
        <div className="stickyHeader">
          <h3 className="homeHeader">FUN TO-DO</h3>
          <AddBookPrompt />
        </div>
        <div className="books">
          <BookList books={this.state.books} />
        </div>
      </div>
    );
  }
}

export default Home;
