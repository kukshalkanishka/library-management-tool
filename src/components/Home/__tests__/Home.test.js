import React from "react";
import { shallow, configure } from "enzyme";
import Home from "../Home";

import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Home", () => {
  it("should render properly", () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
