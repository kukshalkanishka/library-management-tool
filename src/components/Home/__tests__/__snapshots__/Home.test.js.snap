// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render properly 1`] = `
"<div className=\\"todoMainContainer\\">
  <div className=\\"stickyHeader\\">
    <h3 className=\\"homeHeader\\">
      FUN TO-DO
    </h3>
    <AddTodoPrompt />
  </div>
  <div className=\\"todos\\">
    <TodoList todos={{...}} />
  </div>
</div>"
`;
